src/openrct2/peep/GuestPathfinding.cpp

```diff
@@ -1839,6 +1839,45 @@ static void get_ride_queue_end(TileCoordsXYZ& loc)
     loc.z = tileElement->base_height;
 }
 
+/*
+ * If a ride has multiple entrance stations and is set to sync with
+ * adjacent stations, cycle through the entrance stations (based on
+ * number of rides the peep has been on) so the peep will try the
+ * different sections of the ride.
+ * In this case, the ride's various entrance stations will typically,
+ * though not necessarily, be adjacent to one another and consequently
+ * not too far for the peep to walk when cycling between them.
+ * Note: the same choice of station must made while the peep navigates
+ * to the station. Consequently a truly random station selection here is not
+ * appropriate.
+ */
+static StationIndex guest_pathfinding_select_random_station(
+    const Guest* guest, int32_t numEntranceStations, std::bitset<MAX_STATIONS> entranceStations, StationIndex closestStationNum)
```

Comment:
```suggestion
    const Guest* guest, int32_t numEntranceStations, std::bitset<MAX_STATIONS>& entranceStations, StationIndex& closestStationNum)
```
Because you're altering `entranceStations` and `closestStationNum`, you should receive a reference to them instead.

Reply:
No.

entranceStations is not modified after the call to `guest_pathfinding_select_random_station()`, so a reference is not necessary.
The new value of `closestStationNum` is _returned_.

Reply:
You're right, it only assigns to `closestStationNum`, but never reads from it inside the method, so you don't even need to send this parameter if you change the last return to `return 0` and the `closestStationNum = i;` to `return i;`. Did I interpret it correctly?

As for `entranceStations`, I get that it is not being read after the function is called, but if it ever gets to, it'll be hard to debug, unless we don't expect the function to alter the outer bitset at all.

Reply:
OK, addressed both points.